walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Iterator;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) { //Was able to get the runtime down from ~4.5 secs to ~2.8 secs by removing unnecessary loops(condensing the logic), then down to ~2.6 secs by simplifying conditionals
```

Comment:
Nit: Generally speaking, add comment above the block of code it pertains to (like you did on line 36)